### ISSUE TITLE
use this at high traffic causes server hang?

### ISSUE THREAD

**Issue by raitucarp** [mattcg/socks5-http-client#10]
I don't know what happens, but could you run test with scenario high traffic?

Because my server hang and can't restart with PM2. If that is happens with node 0.12.0, I switch to 0.10.34 or 0.10.36, it is still causes hang.. Thanks

ps: the complete indication of my server hang is the first post, FYI:
https://github.com/Unitech/PM2/issues/1072

**Comment by mattcg** [comment]
Thanks for reporting. Could you give a little bit of context please? How many connections per second are you making? Are they all to different hosts?

**Comment by raitucarp** [comment]
the complete description in the first post that I reference above

I use default maxsockets setting of node 0.12.0, which is Infinity. I don't know, this is weird, some times causes server to hang, and can't be kill by PM2, except do it manually kill -9

I don't count the connections per second, because my website is crawled by search engine, and difficult to count how many hits, crawler access my website.

**Comment by mattcg** [comment]
Have you tried setting maxsockets to something low? Does it still hang when you do that?

You could set it to 1000, test, if it still hangs set it to 500, test again, and so on until it stops hanging.

**Comment by raitucarp** [comment]
yes, I tested it, and still down sometimes, also return socket hangsup if I limit maxsockets

Maybe I should tune what the right value of maxsockets, and I will update this issue, thanks
